dicoogle/src/main/java/pt/ua/dicoogle/plugins/PluginController.java

```diff
@@ -603,8 +622,35 @@ public JointQueryTask query(JointQueryTask holder, List<String> querySources, fi
         });
         //logger.info("Prepared Query Task: QueryString");
         return queryTask;
-    }        
- 
+    }
+
+
+    private Task<Iterable<SearchResult>> getTaskForQueryDim(final String querySource, final String query,
+                                                         final DimLevel level, final Object ... parameters){
+
+        final QueryInterface queryEngine = getQueryProviderByName(querySource, true);
+        //returns a tasks that runs the query from the selected query engine
+        String uid = UUID.randomUUID().toString();
+        Task<Iterable<SearchResult>> queryTask = new Task<>(uid, querySource,
+                new Callable<Iterable<SearchResult>>(){
+                    @Override public Iterable<SearchResult> call() throws Exception {
+                        if(queryEngine == null || !(queryEngine instanceof QueryDimInterface)) return Collections.emptyList();

```

Comment:
A more fail-fast approach would be performing this check before the task is created, which should be fine if you're OK with throwing when the program fails to create the task.

Reply:
This is a private method - safer and comfortable to return an empty list. No clear benefit to throw exception IMHO and throw error at this stage.